src/fw/drivers/sf32lb52/temperature.c

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2025 SiFli Technologies(Nanjing) Co., Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "drivers/temperature.h"
+
+#include "bf0_hal.h"
+#include "board/board.h"
+#include "console/prompt.h"
+#include "drivers/gpio.h"
+#include "drivers/periph_config.h"
+#include "drivers/voltage_monitor.h"
+#include "kernel/util/sleep.h"
+#include "mfg/mfg_info.h"
+#include "services/common/regular_timer.h"
+#include "system/logging.h"
+#include "system/passert.h"
+
+static TSEN_HandleTypeDef TSEN_Handle = {.Instance = hwp_tsen};
+
+void temperature_init(void) {
+  if (HAL_TSEN_Init(&TSEN_Handle) != HAL_OK) {
+    PBL_ASSERTN(0);
+  }
+}
+
+static void prv_tsen_enable(TSEN_HandleTypeDef *htsen) {
+  htsen->Instance->TSEN_CTRL_REG &= ~TSEN_TSEN_CTRL_REG_ANAU_TSEN_RSTB;
+  htsen->Instance->TSEN_CTRL_REG |=
+      TSEN_TSEN_CTRL_REG_ANAU_TSEN_EN | TSEN_TSEN_CTRL_REG_ANAU_TSEN_PU;
+  htsen->Instance->TSEN_CTRL_REG |= TSEN_TSEN_CTRL_REG_ANAU_TSEN_RSTB;
+  HAL_Delay_us(20);
+  htsen->Instance->TSEN_CTRL_REG |= TSEN_TSEN_CTRL_REG_ANAU_TSEN_RUN;
+  htsen->State = HAL_TSEN_STATE_ENABLED;
+}
+
+static void prv_tsen_disable(TSEN_HandleTypeDef *htsen) {
+  htsen->Instance->TSEN_CTRL_REG &=
+      ~(TSEN_TSEN_CTRL_REG_ANAU_TSEN_EN | TSEN_TSEN_CTRL_REG_ANAU_TSEN_PU);
+  htsen->State = HAL_TSEN_STATE_READY;
+}
+
+static int32_t prv_temperature_read_milli(void) {
+  int32_t temp = 0;
+
+  uint32_t count = 0;
+  prv_tsen_enable(&TSEN_Handle);
+  while ((TSEN_Handle.Instance->TSEN_IRQ & TSEN_TSEN_IRQ_TSEN_IRSR) == 0) {
+    HAL_Delay(1);
+    count++;
+    if (count > HAL_TSEN_MAX_DELAY) {
+      temp = -HAL_ERROR_TEMPRATURE;
+      break;
+    }
+  }
+  TSEN_Handle.Instance->TSEN_IRQ |= TSEN_TSEN_IRQ_TSEN_ICR;
+  const uint32_t SLOPE_NUM = 2971;
+  const uint32_t SLOPE_DEN = 40;
+  const int32_t OFFSET = 277539;
+  const uint32_t ROUND_ADD = SLOPE_DEN / 2;
+  if (temp >= 0) {
+    uint32_t raw = TSEN_Handle.Instance->TSEN_RDATA;
+    uint32_t D = raw + 3000;
+    uint32_t num = D * SLOPE_NUM + ROUND_ADD;  // 带四舍五入偏移
+    uint32_t tmp = num / SLOPE_DEN;
+    temp = tmp - OFFSET;
+  }
+  prv_tsen_disable(&TSEN_Handle);
+
+  return temp;
+}
+
+int32_t temperature_read(void) { return prv_temperature_read_milli(); }
```

Comment:
just implement the function here, internal static is redundant, in fact, looks like everything could be condensed here.